File: pinyougou/pinyougou-order/pinyougou-order-interface/src/main/java/com/pinyougou/order/service/OrderService.java
package com.pinyougou.order.service;

import com.pinyougou.pojo.TbOrder;
import com.pinyougou.pojo.TbOrderItem;
import com.pinyougou.pojo.TbPayLog;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.ChangeLong;
import com.pinyougou.vo.PageResult;

import java.util.List;

public interface OrderService extends BaseService<TbOrder> {

    PageResult search(Integer page, Integer rows, TbOrder order);

    /**
     *保存订单数据并返回日志id
     * @param order
     * @return
     */
    String addOrder(TbOrder order);

    /**
     * 根据支付日志id查找支付日志信息
     * @param outTradeNo
     * @return
     */
    TbPayLog findPayLogByOutTradeNo(String outTradeNo);

    /**
     * 获得商品描述
     * @param outTradeNo
     * @return
     */
    List<TbOrderItem> findBody(String outTradeNo);

    /**
     * 更新订单状态和信息
     * @param outTradeNo
     * @param transactionId
     */
    void updateOrderStatus(String outTradeNo, String transactionId);

    /**
     * 查询全部订单并处理id过长
     * @return
     */
    List<ChangeLong> findAllOrder();

    /**
     * 查询全部订单并处理id过长(商家)
     * @param order
     * @return
     */
    List<ChangeLong> findAllOrderByUser(TbOrder order);

    /**
     * 修改地址
     * @param
     */
    void updateArea(TbOrder tbOrder,Long id);

    /**
     *状态查询
     * @param value
     * @return
     */
    List<ChangeLong> findStatus(String value);

}

File: pinyougou/pinyougou-manage-web/src/main/webapp/js/service/orderService.js
app.service("orderService", function ($http) {
    //加载列表数据
    this.findAll = function () {
        return $http.get("../order/findAll.do");
    };

    this.findPage = function (page, rows) {
        return $http.get("../order/findPage.do?page=" + page + "&rows=" + rows);
    };

    this.update = function (entity,id) {
        return $http.post("../order/update.do?id="+id, entity);
    };

    this.findOne = function (id) {
        return $http.get("../order/findOne.do?id=" + id);
    };

    this.search = function (page, rows, searchEntity) {
        return $http.post("../order/search.do?page=" + page + "&rows=" + rows, searchEntity);

    };

    this.findAllOrder = function () {
        return $http.post("../order/findAllOrder.do");
    };

    //状态查询
    this.findStatus = function (value) {
        return $http.get("../order/findStatus.do?value="+value);
    };
});